engine/execution/storehouse.go

```diff
@@ -0,0 +1,74 @@
+package execution
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// RegisterStore is the interface for register store
+// see implementation in engine/execution/storehouse/register_store.go
+type RegisterStore interface {
+	// GetRegister first try to get the register from InMemoryRegisterStore, then OnDiskRegisterStore
+	GetRegister(height uint64, blockID flow.Identifier, register flow.RegisterID) (flow.RegisterValue, error)
+
+	// SaveRegisters saves to InMemoryRegisterStore first, then trigger the same check as OnBlockFinalized
+	// Depend on InMemoryRegisterStore.SaveRegisters
+	SaveRegisters(header *flow.Header, registers []flow.RegisterEntry) error
```

Comment:
Should this accept `height` and `blockID` instead of the header? It's not clear to me why it would differ from `GetRegister`. Can we unify? My vote is to have both accept height and blockID so there's less dependency.

Reply:
What are possible errors? add a comment?

Reply:
I see. That makes sense.